src/mode/modeHandler.ts

```diff
@@ -707,6 +707,10 @@ export class ModeHandler implements vscode.Disposable {
     if (result instanceof Position) {
       vimState.cursorPosition = result;
     } else if (isIMovement(result)) {
+      if (result.failed) {
+        vimState.recordedState = new RecordedState();
```

Comment:
Yeah, sorry, RecorededState is an awful name! It used to be called ActionState, which was a little better, but still bad.